### ISSUE TITLE
[pickers] Placeholder accessability issue (low contrast)

### ISSUE THREAD

**Issue by morzel85** [mui/mui-x#20238]
### Steps to reproduce

Steps:
1. Open DatePicker docs page: https://mui.com/x/react-date-pickers/date-picker
2. Find a date field, clear the date value and move focus outside of the field.
3. Inspect element in DevTools and notice the contrast ratio (axe accessibility plugin also detects the issue):

<img width="1582" height="987" alt="Image" src="https://github.com/user-attachments/assets/11247c30-3816-4e9c-ad49-8c77ffdc38fb" />

### Current behavior

Placeholders have insufficient contrast which is due to low opacity:

<img width="1399" height="544" alt="Image" src="https://github.com/user-attachments/assets/edf482f9-ef60-4a0b-9ec2-9e6bd7be1bbd" />

### Expected behavior

Placeholders should have contrast that does not violate accessibility guidelines.

Overriding placeholder color/opacity should be easier with themes (it's problematic because opacity override needs to take into account if field is focused and filled in).

MUI demo pages should be free of accessibility issues detectable by a quick scan with axe DevTools.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: Linux 6.14 Ubuntu 24.04.3 LTS 24.04.3 LTS (Noble Numbat)
  Binaries:
    Node: 22.17.0 - /home/morzel/.nvm/versions/node/v22.17.0/bin/node
    npm: 10.9.2 - /home/morzel/.nvm/versions/node/v22.17.0/bin/npm
    pnpm: 10.7.0 - /home/linuxbrew/.linuxbrew/bin/pnpm
  Browsers:
    Chrome: 142.0.7444.59
    Firefox: 144.0.2
```
</details>

**Search keywords**: contrast, placeholder

**Comment by michelengelen** [comment]
Hi @morzel85 and thanks for raising this.
I did also notice that the color differs for the label and the placeholder.
The label has `rgba(0,0,0,0.6)`, or `#666666` on white background, resulting in a contrast score of 5.74, which is ok (not great though), whereas the placeholder has `rgba(0,0,0,0.87)` with an additional opacity of 0.42, or `#a2a2a2` on white background, which is resulting in a contrast score of only 2.55.

Proposal is to adjust the coloring of the placeholder to use the same values as the label.

Objections @flaviendelangle @kenanyusuf ?

**Comment by kenanyusuf** [comment]
The computed placeholder color (on white) is consistent with the placeholder styling in the MUI Text Field. I agree that increasing the contrast would improve accessibility, but I think this should be addressed at the design system level so it remains consistent across components.

**Comment by michelengelen** [comment]
The placeholder uses a hardcoded value when no vars are present: https://github.com/mui/mui-x/blob/ee5fe84cb808fcecb826e63042038ff38185c05b/packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx#L97-L103

... and `currentColor`: https://github.com/mui/mui-x/blob/ee5fe84cb808fcecb826e63042038ff38185c05b/packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx#L85-L88

Researching this I found that the `InputBase` component in material-ui also uses the same approach: https://github.com/mui/material-ui/blob/c08f4695073f183f0332c402a1504e03f05deccc/packages/mui-material/src/InputBase/InputBase.js#L150-L172

@siriwatknp did this pop up before as well? Seems like the placeholder is always handled differently to the label.

**Comment by michelengelen** [comment]
grooming note: the contrast is customizable through theming, so we will provide a workaround for now and see if the issue is still relevant after that before closing it. The additional opacity added could be a way to handle weird browser defaults when autofilling.

**Comment by siriwatknp** [comment]
We don't have plan to change the opacity value any soon.
The workaround is to use CSS theme variables feature and update the theme token like this:

```js
const theme = createTheme({
  cssVariables: true,
  colorSchemes: {
    light: {
      opacity: {
        inputPlaceholder: 0.6,
      },
    },
    dark: {
      opacity: {
        inputPlaceholder: 0.6,
      },
    },
  },
});
```

For full workaround, see https://stackblitz.com/edit/r2lgcc9h?file=src%2FDemo.tsx

**Comment by morzel85** [comment]
**Edit: I actually had to revert this patch. It messed up color theme (light/dark mode) switching!**

Thanks for the snippet, I've used the `opacity.inputPlaceholder` tweak and it works (but I bumped it to 0.65 because 6 was still below the recommend minimum).

Would be nice though to have it done in MUI directly, one of the reasons we pick MUI is that building your our own accessible component suite is a lot of work. It's a bit hard to pitch for MUI if a demo page has a11y issues.
